src/main/java/budgetflow/command/LogExpenseCommand.java

```diff
@@ -0,0 +1,153 @@
+import budgetflow.exception.MissingCategoryException;
+import budgetflow.exception.MissingDescriptionException;
+import budgetflow.exception.MissingExpenseException;
+import budgetflow.expense.Expense;
+import budgetflow.expense.ExpenseList;
+import budgetflow.income.Income;
+import budgetflow.parser.DateValidator;
+
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.logging.Logger;
+
+//@@author thienkimtranhoang
+public class LogExpenseCommand extends Command{
+    public static final String ERROR_INVALID_DATE = "Error: Date is not a valid date";
+
+    private static final Logger logger = Logger.getLogger(LogExpenseCommand.class.getName());
+    private static final String LOG_EXPENSE_COMMAND_PREFIX = "log-expense ";
+    private static final int LOG_EXPENSE_COMMAND_PREFIX_LENGTH = LOG_EXPENSE_COMMAND_PREFIX.length();
+
+    private static final String ERROR_EMPTY_EXPENSE = "Expense should not be empty";
+    private static final String ERROR_MISSING_EXPENSE_CATEGORY = "Error: Expense category is required.";
+    private static final String ERROR_MISSING_EXPENSE_DESCRIPTION = "Error: Expense description is required.";
+    private static final String ERROR_MISSING_EXPENSE_AMOUNT = "Error: Expense amount is required.";
+    private static final String ERROR_MISSING_EXPENSE_DATE = "Error: Expense date is required.";
+    private static final String ERROR_INCORRECT_EXPENSE_DATE = "Error: Income date is in wrong format." +
+            "please use DD-MM-YYYY format.";
+
+    public LogExpenseCommand(String input) {
+        super(input);
+        this.commandType = CommandType.CREATE;
+    }
+
+    /**
+     * Log new user expense into the expense list
+     * @param expenseList the list storing all expenses
+     * @throws MissingDateException if user miss the date of expense or date tag
+     * @throws InvalidNumberFormatException if amount does not follow valid number format
+     * @throws MissingAmountException if user miss the amount of expense or amount tag
+     * @throws MissingCategoryException if user miss the category of expense or expense tag
+     * @throws MissingDescriptionException if user miss description of expense or expense tag
+     */
+    @Override
+    public void execute(List<Income> incomes, ExpenseList expenseList) throws MissingDateException,
+            InvalidNumberFormatException, MissingAmountException, MissingCategoryException,
+            MissingDescriptionException, MissingExpenseException {
+        Expense expense = extractExpense(input);
+        expenseList.add(expense);
+        this.outputMessage = "Expense logged: " + expense.getCategory() + " | " + expense.getDescription() +
+                " | $" + String.format("%.2f", expense.getAmount()) + " | " + expense.getDate();
+    }
+
+    //@@author dariusyawningwhiz
+    private Expense extractExpense (String input) throws InvalidNumberFormatException,
```

Comment:
This method is really long (>30LOC) consider refactoring it into multiple methods.